src/index.js

```diff
@@ -49,19 +49,21 @@ async function start(fields) {
   })
 
   try {
+    const accountInfo = await googleHelper.getAccountInfo()
     const contacts = await googleHelper.getAllContacts({
       personFields: FIELDS.join(',')
     })
-    const ioCozyContacts = contacts.map(transpile.toCozy)
-    const newIoCozyContacts = await filterData(
-      ioCozyContacts,
-      'io.cozy.contacts',
-      {
-        keys: ['email']
+    const ioCozyContacts = contacts.map(transpile.toCozy).map(contact => ({
+      ...contact,
+      metadata: {
+        google: {
+          ...contact.metadata['google'],
+          from: accountInfo.emails[0].value
+        }
       }
-    )
-    addData(newIoCozyContacts, 'io.cozy.contacts')
+    }))
+    updateOrCreate(ioCozyContacts, 'io.cozy.contacts', ['resourceName'])
```

Comment:
A return here would prevent the Promise result to be lost in limbo. It is necessary so that the BaseKonnector can correctly catch errors.

Reply:
TIL, I'll update this code line.